### PR TITLE
docs(CL): Swaps

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #XXX

## What is the purpose of the change

Documenting our concentrated liquidity swaps from discussions with @jonator .

It includes the high-level explanation and logic done in the swap and adds an example in the appendix, helping to visualize the swap done over a tick range with 2 positions.

## Testing and Verifying

This change is a trivial rework / code cleanup without any test coverage.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? not applicable

### CODE REVIEW

File: x/concentrated-liquidity/swaps.go [hunk 1]
```diff
@@ -14,13 +14,42 @@ import (
...
+
+	// Current sqrt price while calculating swap.
+	// Initialized to the pool's current sqrt price.
+	// Updated after every swap step.
+	sqrtPrice sdk.Dec
+	// Current tick while calculating swap.
+	// Initialized to the pool's current tick.
+	// Updated each time a tick is crossed.
+	tick sdk.Int
+	// Current liqudiity within the active tick.
```
Comment (AlpinYukseloglu) [inline x/concentrated-liquidity/swaps.go:44]:
```suggestion
	
	// Current liqudiity within the active tick.
```

File: x/concentrated-liquidity/README.md [hunk 2]
```diff
@@ -472,31 +472,275 @@ func (k Keeper) withdrawPosition(
 
 > As a trader, I want to be able to swap over a concentrated liquidity pool so that my trades incur lower slippage
 
-Unlike balancer pools where liquidity is spread out over an infinite range, concentrated liquidity pools allow for deeper liquidity at the current price, which in turn allows trades the incur less slippage.
+Unlike balancer pools where liquidity is spread out over an infinite range, concentrated liquidity pools allow for deeper liquidity at the current price, which in turn allows trades to incur less slippage.
```
Comment (AlpinYukseloglu) [inline x/concentrated-liquidity/README.md:475]:
```suggestion
Unlike balancer pools where liquidity is spread out over an infinite range, concentrated liquidity pools allow for LPs to provide deeper liquidity for specific price ranges, which in turn allows traders to incur less slippage on their trades.
```

**Review by AlpinYukseloglu — APPROVED** [review]
Looks good on an initial skim. Just as an FYI I did not review the mechanics of computation section in great detail. Thanks for adding this!

File: x/concentrated-liquidity/README.md [hunk 3]
```diff
@@ -472,31 +472,275 @@ func (k Keeper) withdrawPosition(
...
+than enough to cross tick 0 and it returns 10_000 of token zero out while consuming half of
+token one in (2500).
+
+Now, we update the swap state as follows:
+
+- `amountSpecifiedRemaining` is set to 5000 - 2_500 = 2_500 tokens one in remaining.
+
+- `amountCalculated` is set to 10_000 tokens zero out calculated.
+
+- `sqrtPrice` is set to the sqrt price of the crossed initialzed tick 0 (0).
```
Comment (jonator) [inline x/concentrated-liquidity/README.md:708]:
```suggestion
- `sqrtPrice` is set to the sqrt price of the crossed initialized tick 0 (0).
```

File: x/concentrated-liquidity/README.md [hunk 4]
```diff
@@ -472,31 +472,275 @@ func (k Keeper) withdrawPosition(
...
+
+- `amountCalculated` is set to 10_000 + 12_500 = 22_500 tokens zero out calculated.
+
+- `sqrtPrice` is set to the reached sqrt price.
+
+- `tick` is set to an uninitialized tick associated with the reached sqrt price (70).
+
+- `liquidity` is set kept the same as we did not cross any initialized tick.
+
+- `feeGrowthGlobal` is set to 0 because we assumed zero swap fee.
```
Comment (jonator) [inline x/concentrated-liquidity/README.md:736]: I think swap fee would also be valuable to add in a new section. At least describing how the swap fee gets distributed to the positions after a successful swap.

Comment (p0mvn) [reply x/concentrated-liquidity/README.md]: Updated: [1bd2f8d](https://github.com/osmosis-labs/osmosis/pull/4696/commits/1bd2f8d73184de3dec1b8db63304be1d84b9e658)

Planning to also write up a better section on fees separately

**Pull request merged**